librecad/src/lib/engine/rs_units.cpp

```diff
@@ -228,54 +228,54 @@ QString RS_Units::unitToString(RS2::Unit u, bool t) {
 /**
  * Converts a string into a unit enum.
  */
-RS2::Unit RS_Units::stringToUnit(const QString& u) {
-    RS2::Unit ret = RS2::None;
-
-    if (u=="None") {
-        ret = RS2::None;
-    } else if (u==QObject::tr("Inch")) {
-        ret = RS2::Inch;
-    } else if (u==QObject::tr("Foot")) {
-        ret = RS2::Foot;
-    } else if (u==QObject::tr("Mile")) {
-        ret = RS2::Mile;
-    } else if (u==QObject::tr("Millimeter")) {
-        ret = RS2::Millimeter;
-    } else if (u==QObject::tr("Centimeter")) {
-        ret = RS2::Centimeter;
-    } else if (u==QObject::tr("Meter")) {
-        ret = RS2::Meter;
-    } else if (u==QObject::tr("Kilometer")) {
-        ret = RS2::Kilometer;
-    } else if (u==QObject::tr("Microinch")) {
-        ret = RS2::Microinch;
-    } else if (u==QObject::tr("Mil")) {
-        ret = RS2::Mil;
-    } else if (u==QObject::tr("Yard")) {
-        ret = RS2::Yard;
-    } else if (u==QObject::tr("Angstrom")) {
-        ret = RS2::Angstrom;
-    } else if (u==QObject::tr("Nanometer")) {
-        ret = RS2::Nanometer;
-    } else if (u==QObject::tr("Micron")) {
-        ret = RS2::Micron;
-    } else if (u==QObject::tr("Decimeter")) {
-        ret = RS2::Decimeter;
-    } else if (u==QObject::tr("Decameter")) {
-        ret = RS2::Decameter;
-    } else if (u==QObject::tr("Hectometer")) {
-        ret = RS2::Hectometer;
-    } else if (u==QObject::tr("Gigameter")) {
-        ret = RS2::Gigameter;
-    } else if (u==QObject::tr("Astro")) {
-        ret = RS2::Astro;
-    } else if (u==QObject::tr("Lightyear")) {
-        ret = RS2::Lightyear;
-    } else if (u==QObject::tr("Parsec")) {
-        ret = RS2::Parsec;
-    }
-
-    return ret;
+RS2::Unit RS_Units::stringToUnit(QString inputString)
+{
+    inputString = inputString.toLower();
+
+    if (inputString == QObject::tr("none"))         return RS2::None;
```

Comment:
I have used multiple `if`s and `return`s because I was suggested that this way doing doing things leads to an _early_ `return`